queue.go

```diff
@@ -94,18 +119,23 @@ type Queue struct {
 }
 
 // NewQueue creates and returns a new Queue structure
-func NewQueue(url string, name string, prefetchSize int, isConsumer, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+func NewQueue(name string, prefetchSize int, isConsumer, isExchange, durable bool, jobs chan amqp.Delivery) (*Queue, error) {
+	if cfg.HostName == "" || cfg.Port == "" ||
+		cfg.UserName == "" || cfg.Password == "" {
+		panic("hostname,port,username and password are required for establishing the connection")
+	}
+
 	q := &Queue{
```

Comment:
This doesn't have to be a queue because it can be an exchange so I would rename the struct to something more general. How about Channel?